url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/parser/Parser.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * Scripty Programming Language
+ * Copyright (C) 2010-2012 Bruno Ranschaert, S.D.I.-Consulting BVBA
+ * http://www.sdi-consulting.be
+ * mailto://info@sdi-consulting.be
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.sdicons.scripty.parser;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * A minimal parser. Only lists, strings (delimited or not) and pairs (to make named parameters possible).
+ * It is not intended as a full programming language, it is intended as a command processor.
+ * No numbers, floats, symbols or other built-in data structures. The command extensions have to provide these 
+ * if needed. The pairs are a special feature, it is a syntactic convenience to allow parser support
+ * for key/value pairs which are often used to pass command line options, it has nothing to do with conses.
+ * <p>
+ * Examples:
+ * <pre>
+ * abc           -> A string.
+ * "abc def"     -> A string with embedded whitespace.
+ * key=val       -> A pair.
+ * ()            -> Empty list.
+ * (a b c d)     -> List of four strings.
+ * (a b (c (d))) -> Nested lists.
+ * (a b)=(c d)   -> A pair where both key and value are lists.
+ * a = b = c     -> pair(pair(a,b),c). So pairing works from left to right.
+ * ; comments    -> Comments start with a ; and are skipped by the lexer.
+ * </pre>
+ *
+ */
+public class Parser
+{
+    private Token pushback = null;
+    
+    /**
+     * Lexer method.
+     * 
+     * @param aBuffer
+     * @return
+     */
+    private Token getNextToken(IParserInput aBuffer)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.